### PR TITLE
PROD-352: configured eslint, prettier and husky

### PR DESCRIPTION
**Pull request by yash431garg**
- Description
Integrated eslint for lint error includes tslint, import and hooks warning. Added prettier with formatting rules and husky for precommit check
- What are the steps to test that this code is working?
No functional changes
- Screen shots or recordings for UI changes
N/A

### CODE REVIEW

File: .prettierignore [hunk 1]
```diff
@@ -0,0 +1,10 @@
+node_modules/
+.next/
+public/
+out/
+coverage/
+build/
+.vercel/
+docs/
+prisma/migrations/
+
```
Comment (neel-ds) [inline .prettierignore:10]: You can add more folders and files in ignore like:
- all files / folders starting with .
- yarn.lock
- postgres-data
- README files

File: .storybook/README.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ### Storybook
 
 When creating UI components developer should take care to run the command:
-```yarn run storybook```
+`yarn run storybook`
```
Comment (neel-ds) [inline .storybook/README.md:4]: This change should be reverted once you follow up on `prettierignore` review

File: app/components/Icons/DeckIcon.tsx [hunk 3]
```diff
@@ -1,7 +1,6 @@
 import { ICON_DEFAULT_WIDTH_HEIGHT, IconProps } from ".";
 
 export function DeckIcon({
-  fill = "#fff",
```
Comment (neel-ds) [inline app/components/Icons/DeckIcon.tsx:4]: Can we keep this fill prop unless by default is #fff? Not sure if it was intentional

**Review by neel-ds — CHANGES_REQUESTED** [review]
Kindly add more files and folders in `prettierignore` as mentioned below and revert the changes of same files or folders. I also found line breaks between imports in almost every files. What's the purpose behind it? Please remove that rule unless there is convention

**Comment by yash431garg** [comment]
> Kindly add more files and folders in `prettierignore` as mentioned below and revert the changes of same files or folders. I also found line breaks between imports in almost every files. What's the purpose behind it? Please remove that rule unless there is convention

The reason behind line breaks in import was to separate packages which are external and internal components.

File: app/components/Icons/DeckIcon.tsx (hunk 3 above)
Comment (yash431garg) [reply app/components/Icons/DeckIcon.tsx:4]: It was not used here that why comes to property of unused vars.

File: .husky/pre-commit [hunk 4]
```diff
@@ -0,0 +1,25 @@
+#!/bin/sh
```
Comment (marvinmarnold) [inline .husky/pre-commit:1]: Will all of these also be run when a PR is open?

Comment (yash431garg) [reply .husky/pre-commit:1]: No, It will run after we commits and will block the commit if there is some issue.

Comment (yash431garg) [reply .husky/pre-commit:1]: Sure, I will add this in the workflow.

**Comment by marvinmarnold** [comment]
@neel-ds can you please re-review?

**Review by marvinmarnold — APPROVED** [review]

**Review by marvinmarnold — DISMISSED** [review]

File: .prettierignore (hunk 1 above)
Comment (neel-ds) [reply .prettierignore:14]: To ignore README files, `*.md` use this instead of README*

**Review by neel-ds — DISMISSED** [review]

**Review by marvinmarnold — APPROVED** [review]

**Review by neel-ds — APPROVED** [review]

**Review by marvinmarnold — APPROVED** [review]

**Pull request merged**